x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -210,9 +210,10 @@ export const AlertsTableComponent: FC<DetectionEngineAlertTableProps> = ({
     return undefined;
   }, [isEventRenderedView]);
 
-  const dataTableStorage = getDataTablesInStorageByIds(storage, [TableId.alertsOnAlertsPage]);
```

Comment:
@logeekal this may be a much much more involved change from what I'm seeing. From what I can tell only pulling in the initial columns like this is necessary for the table to properly own control to set the columns, but haven't dug into exactly what the table is doing under the hood to manage these changes.  My changes here break the tests, because the expected table updates on load aren't happening properly on the Rule details page, even though the `tableId` should account for the table settings for that table. Returning this to how it was previously fixes the tests, but, since localStorage updates aren't listened to, the security_solution column toggle functionality doesn't work. Also fyi @kqualters-elastic

Reply:
Hey @michaelolo24, I am not sure I fully understand but I tested it in rule details page and it seems to be working fine. I see that you have removed the hard coding of `TableId.alertsOnAlertsPage` , which is great. Could you please point me to exact scenarios that fail.